### PR TITLE
Deprecate FigureCanvas.{get,set}_window_title.

### PR DESCRIPTION
**Pull request by anntzer**
This really belongs on the manager, and is a mild footgun on the canvas.

See discussion at https://github.com/matplotlib/matplotlib/issues/17721.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/backend_bases.py [hunk 1]
```diff
@@ -2228,6 +2228,8 @@ def get_default_filetype(cls):
         """
         return rcParams['savefig.format']
 
+    @cbook.deprecated(
+        "3.3", alternative="manager.get_window_title or GUI-specific methods")
```
Comment (QuLogic) [inline lib/matplotlib/backend_bases.py:2232]:
```suggestion
        "3.4", alternative="manager.get_window_title or GUI-specific methods")
```

File: lib/matplotlib/backend_bases.py [hunk 2]
```diff
@@ -2236,6 +2238,8 @@ def get_window_title(self):
         if self.manager:
             return self.manager.get_window_title()
 
+    @cbook.deprecated(
+        "3.3", alternative="manager.get_window_title or GUI-specific methods")
```
Comment (QuLogic) [inline lib/matplotlib/backend_bases.py:2242]:
```suggestion
        "3.4", alternative="manager.set_window_title or GUI-specific methods")
```

**Comment by anntzer** [comment]
oops, fixed

**Comment by timhoffm** [comment]
CI failure: Tests need to account for the deprecation.

**Comment by anntzer** [comment]
yup

File: lib/matplotlib/backend_bases.py (hunk 2 above)
Comment (QuLogic) [reply lib/matplotlib/backend_bases.py]: The name is still wrong here.

File: lib/matplotlib/backend_bases.py [hunk 3]
```diff
@@ -2236,6 +2238,8 @@ def get_window_title(self):
         if self.manager:
             return self.manager.get_window_title()
 
+    @cbook.deprecated(
+        "3.4", alternative="manager.get_window_title or GUI-specific methods")
```
Comment (dstansby) [inline lib/matplotlib/backend_bases.py:2242]:
```suggestion
        "3.4", alternative="manager.set_window_title or GUI-specific methods")
```

Comment (anntzer) [reply lib/matplotlib/backend_bases.py]: oops, fixed

**Comment by anntzer** [comment]
rebased

**Pull request merged**